mlflow_extensions/serving/engines/huggingface_utils.py

```diff
@@ -21,6 +23,47 @@ def snapshot_download_local(repo_id: str, local_dir: str, tokenizer_only: bool =
     return model_local_path
 
 
+def get_local_snapshot(repo_id, cache_dir, local_dir):
+    """
+    Scans local_dir for a model matching the repo_id and returns the path or None.
+    Also supports non-cache directories.
+    """
+    snapshot_path = None
+
+    try:
+        scan = scan_cache_dir(cache_dir=cache_dir)
+        for repo in [r for r in scan.repos if r.repo_id == repo_id]:
+            for rev in repo.revisions:
+                snapshot_path = rev.snapshot_path
+                break
+            if snapshot_path:
+                break
+
+    except Exception as e:
+        print(f"Cache directory scan failed: {e}")
+
+        # If scan_cache_dir doesn't work, check if the directory contains config.json
+        if os.path.exists(local_dir):
+            config_path = os.path.join(local_dir, "config.json")
+            if os.path.exists(config_path):
+                snapshot_path = local_dir
+
+        if not snapshot_path:
+            print("No valid model directory found.")
```

Comment:
for all print can we use debug_msg()... ideally at a point we can switch it with a logger that is unbuffered and rotated into volumes.